Riot/Modules/Room/EditHistory/EditHistoryViewController.swift

```diff
@@ -0,0 +1,196 @@
+    private var errorPresenter: MXKErrorPresentation!
+    private var activityPresenter: ActivityIndicatorPresenter!
+
+    // MARK: - Setup
+    
+    class func instantiate(with viewModel: EditHistoryViewModelType) -> EditHistoryViewController {
+        let viewController = StoryboardScene.EditHistoryViewController.initialScene.instantiate()
+        viewController.viewModel = viewModel
+        viewController.theme = ThemeService.shared().theme
+        return viewController
+    }
+    
+    // MARK: - Life cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Do any additional setup after loading the view.
+        
+        self.title = "Edits history"
+        
+        self.setupViews()
+        self.keyboardAvoider = KeyboardAvoider(scrollViewContainerView: self.view, scrollView: self.scrollView)
+        self.activityPresenter = ActivityIndicatorPresenter()
+        self.errorPresenter = MXKErrorAlertPresentation()
+        
+        self.registerThemeServiceDidChangeThemeNotification()
+        self.update(theme: self.theme)
+        
+        self.viewModel.viewDelegate = self
+
+        self.viewModel.process(viewAction: .loadMore)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        self.keyboardAvoider?.startAvoiding()
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        
+        self.keyboardAvoider?.stopAvoiding()
+    }
+    
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return self.theme.statusBarStyle
+    }
+    
+    // MARK: - Private
+    
+    private func update(theme: Theme) {
+        self.theme = theme
+        
+        self.view.backgroundColor = theme.headerBackgroundColor
+        
+        if let navigationBar = self.navigationController?.navigationBar {
+            theme.applyStyle(onNavigationBar: navigationBar)
+        }
+
+
+        // TODO:
+        self.messageLabel.textColor = theme.textPrimaryColor
+
+        self.loadMoreButton.backgroundColor = theme.backgroundColor
+        theme.applyStyle(onButton: self.loadMoreButton)
+    }
+    
+    private func registerThemeServiceDidChangeThemeNotification() {
+        NotificationCenter.default.addObserver(self, selector: #selector(themeDidChange), name: .themeServiceDidChangeTheme, object: nil)
+    }
+    
+    @objc private func themeDidChange() {
+        self.update(theme: ThemeService.shared().theme)
+    }
+    
+    private func setupViews() {
+        let closeBarButtonItem = MXKBarButtonItem(title: "Close", style: .plain) { [weak self] in
+            self?.closeButtonAction()
+        }
+        
+        self.navigationItem.rightBarButtonItem = closeBarButtonItem
+        
+        self.scrollView.keyboardDismissMode = .interactive
+        
+        self.messageLabel.text = "VectorL10n.editHistoryTitle"
```

Comment:
Double quotes removal seems to have been forgotten

Reply:
Indeed. I authorised you to remove all of this when you attack the final design :)